q2_longitudinal/_utilities.py

```diff
@@ -650,9 +674,71 @@ def _nmit(table, sample_md, individual_id_column, corr_method="kendall",
     return _dist
 
 
+def _first_differences(metadata, state_column, individual_id_column, metric,
+                       replicate_handling='error', distance_matrix=None):
+
+    # let's force states to be numeric
+    _validate_is_numeric_column(metadata, state_column)
+
+    # create dummy group column in metadata so we can use downstream functions
+    # that split metadata by groups without actually bothering to do so.
+    group_column = "dummy_group"
```

Comment:
you are right... I cannot think of an easy workaround since we need to create this dummy column in the same dataframe. It should not be an issue unless if "dummy_group" is the name of `state_column`, `individual_id_column`, or `metric`.

The best solution I can think of right now is to rename this to: `"a_name_that_no_user_of_qiime2_could_conceivably_ever_use"`

👍 👎 ❓

Reply:
What about calling `uuid.uuid4()` until you find a UUID that isn't a column name in the table? It's hacky but will never collide with a user's columns. We're taking this approach in `Metadata.ids()`, you could reuse that implementation here:

https://github.com/qiime2/qiime2/blob/master/qiime2/metadata.py#L294

Reply:
thanks @jairideout ! I love this solution!!!